src/Bicep.Core/Emit/ParamsFileWriter.cs

```diff
@@ -0,0 +1,153 @@
+            {      
+                case BooleanLiteralSyntax booleanLiteralSyntax:
+                {
+                    jsonWriter.WriteValue(booleanLiteralSyntax.Value);
+                    break;
+                }
+                case IntegerLiteralSyntax integerLiteralSyntax:
+                {
+                    jsonWriter.WriteValue(integerLiteralSyntax.Value);
+                    break;
+                }
+                case StringSyntax stringSyntax:
+                {
+                    jsonWriter.WriteValue(string.Join("", stringSyntax.SegmentValues));
+                    break;
+                }
+                case ObjectSyntax objectSyntax:
+                {
+                    jsonWriter.WriteStartObject();
+                    EmitObjectProperties(objectSyntax, jsonWriter);
+                    jsonWriter.WriteEndObject();
+                    break;
+                }
+                case ArraySyntax arraySyntax:
+                {
+                    jsonWriter.WriteStartArray();
+                    foreach (ArrayItemSyntax itemSyntax in arraySyntax.Items)
+                    {
+                        EmitExpression(itemSyntax.Value, jsonWriter);
+                    }
+                    jsonWriter.WriteEndArray();
+                    break;
+                }
+                case NullLiteralSyntax _:
+                {
+                    jsonWriter.WriteNull();
+                    break;
+                }
+            }
+        }
+
+        public void EmitObjectProperties(ObjectSyntax objectSyntax, JsonTextWriter jsonWriter/*, ISet<string>? propertiesToOmit = null*/)
+        {
+            // var propertyLookup = objectSyntax.Properties.ToLookup(property => property.Value is ForSyntax);
+
+            // // emit loop properties first (if any)
+            // if (propertyLookup.Contains(true))
+            // {
+            //     // we have properties whose value is a for-expression
+            //     this.EmitCopyProperty(() =>
+            //     {
+            //         this.writer.WriteStartArray();
+
+            //         foreach (var property in propertyLookup[true])
+            //         {
+            //             var key = property.TryGetKeyText();
+            //             if (key is null || property.Value is not ForSyntax @for)
+            //             {
+            //                 // should be caught by loop emit limitation checks
+            //                 throw new InvalidOperationException("Encountered a property with an expression-based key whose value is a for-expression.");
+            //             }
+
+            //             this.EmitCopyObject(key, @for, @for.Body);
+            //         }
+
+            //         this.writer.WriteEndArray();
+            //     });
+            // }
+
+            foreach (ObjectPropertySyntax propertySyntax in objectSyntax.Properties)
+            {
+                switch(propertySyntax.Key)
```

Comment:
Is this okay should I do `propertySyntax.TryGetKeyText()` here instead?

Reply:
Yeah, I'd use that method and then throw if you get back `null`. It'll only return `null` in the case you're dealing with string interpolation - which you won't be able to support until expressions are implemented.